### PR TITLE
Apply suggestions from Clippy

### PR DESCRIPTION
**Pull request by imobachgs**
Since the beginning of the project, we have not paid enough attention to Clippy's suggestions. This PR applies most of them

* [bind_instead_of_map](https://rust-lang.github.io/rust-clippy/master/index.html#bind_instead_of_map)
* [clone_double_ref](https://rust-lang.github.io/rust-clippy/master/index.html#clone_double_ref)
* [collapsible_else_if](https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_else_if)
* [derivable_impls](https://rust-lang.github.io/rust-clippy/master/index.html#derivable_impls)
* [expect_fun_call](https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call)
* [match_like_matches_macro](https://rust-lang.github.io/rust-clippy/master/index.html#match_like_matches_macro)
* [needless_borrow](https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow)
* [needless_lifetimes](https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes)
* [needless_return](https://rust-lang.github.io/rust-clippy/master/index.html#needless_return)
* [needless_update](https://rust-lang.github.io/rust-clippy/master/index.html#needless_update)
* [option_as_ref_deref](https://rust-lang.github.io/rust-clippy/master/index.html#option_as_ref_deref)
* [ptr_arg](https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg)
* [redundant_clone](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone)
* [redundant_closure](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure)
* [redundant_pattern_matching](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pattern_matching)
* [search_is_some](https://rust-lang.github.io/rust-clippy/master/index.html#search_is_some)
* [single_char_pattern](https://rust-lang.github.io/rust-clippy/master/index.html#single_char_pattern)
* [single_component_path_imports](https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports)
* [to_string_in_format_args](https://rust-lang.github.io/rust-clippy/master/index.html#to_string_in_format_args)
* [type_complexity](https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity)
* [unit_arg](https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg)
* [unused_io_amount](https://rust-lang.github.io/rust-clippy/master/index.html#unused_io_amount)
* [unused_unit](https://rust-lang.github.io/rust-clippy/master/index.html#unused_unit)
* [useless_conversion](https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion)
* [useless_format](https://rust-lang.github.io/rust-clippy/master/index.html#useless_format)

However, some of them are still pending:

* [await_holding_lock](https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock)
* [type_complexity](https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity)

### CODE REVIEW

File: rust/agama-locale-data/src/timezone_part.rs [hunk 1]
```diff
@@ -42,22 +42,22 @@ impl TimezoneIdParts {
...
 
     fn translate_timezone(&self, mapping: &HashMap<String, String>, timezone: &str) -> String {
         timezone
-            .split("/")
+            .split('/')
             .map(|tzp| {
                 mapping
                     .get(&tzp.to_string())
-                    .expect(format!("Unknown timezone part {tzp}").as_str())
+                    .unwrap_or_else(|| panic!("Unknown timezone part {tzp}"))
```
Comment (mvidner) [inline rust/agama-locale-data/src/timezone_part.rs:60]: JFYI: I liked the original more so I ran clippy and found the reason for the change: the `format!` would always be called.

**Review by mvidner — APPROVED** [review]
Thanks!

For the newbies like me: call it like `cargo clippy`

File: rust/agama-locale-data/src/timezone_part.rs (hunk 1 above)
Comment (imobachgs) [reply rust/agama-locale-data/src/timezone_part.rs:60]: Yes, same with me. :)

**Review by mvidner — APPROVED** [review]
still LGTM

**Pull request merged**